### PR TITLE
modify markduplicates module

### LINKED ISSUE #680: [FEATURE] Allow optional index output for picard/markduplicates
## Is your feature request related to a problem? Please describe

I would like to have an option to emit `.bai` index from MarkDuplicates. There is a `CREATE_INDEX` option for the program and that will generate a .bai index. The module doesn't currently support that optional emission.

## Describe the solution you'd like

Add an optional emit for the `.bai` index.

### PR DESCRIPTION
**Pull request by projectoriented**
## PR checklist

Closes #680  

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `<SOFTWARE>.version.txt` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Review by jfy133 — CHANGES_REQUESTED** [review]
Looks good to me in principle. You could maybe mention in the output description the `bai` is optional.

Also the tests all failed, we would need these passing before approval. could you investigate?

**Comment by projectoriented** [comment]
@jfy133 Thanks! I've updated the description in the meta config 😄  & all tests passed - yay! 🚀

**Review by jfy133 — APPROVED** [review]

**Pull request merged**
